src/libcore/num/f32.rs

```diff
@@ -474,4 +475,85 @@ impl f32 {
         // It turns out the safety issues with sNaN were overblown! Hooray!
         unsafe { mem::transmute(v) }
     }
+
+    /// A 3-way version of the IEEE `totalOrder` predicate.
+    ///
+    /// Because of negative zero and NaNs, the ordinary comparison operators
+    /// for `f32` do not represent a total order.  This method, however,
+    /// defines an ordering between all distinct bit patterns:
+    ///
+    /// - For normal numbers, the ordering matches the comparison operators.
+    /// - Zeros are between the positive and negative normal numbers, with
+    ///   the positive zero greater than the negative zero.
+    /// - Positive infinity is greater than all normal numbers; negative
+    ///   infinity is less than all normal numbers.
+    /// - Positive NaNs are greater than all other values, ordered amongst
+    ///   themselves by their payloads; Negative NaNs are less than all other
```

Comment:
> maybe even use a different implementation on Mips to respect its interpretation of the signaling bit

It seems that the `totalOrder` operation got introduced by the 2008 edition of IEEE 754, i.e. it wasn't present in the 1985 edition. The `totalOrder` operation is specified in terms of the payload and the signaling bit, and not the signifigand bits. If we implement the operation on older MIPS, which is only compliant with the 1985 edition, we are basically "backporting" the definition in the 2008 standard to the 1985 standard. If we don't adjust the parts about the signaling bit, I think this goes really wrong: I think that the spec authors have carefully made the `totalOrder` rules so that they can be implemented by a simple operation on the binary representation. The `totalOrder` spec was made with the 2008 definition in mind, not with the interpretation of 1985 that MIPS chose. We should follow that spirit on MIPS as well and take the implementation that is a simple operation not a "if this is NAN then flip this bit and then do $op" one. 

I don't think that @rkruppe is considering this *too* seriously, but just wanted to say this.

And yeah, we should document this choice about the MIPS+NaN behaviour and maybe say that it might be changed it in the future, to keep options open.